pkg/storage/bucket/user_bucket_client.go

```diff
@@ -9,6 +9,13 @@ import (
 	"github.com/thanos-io/thanos/pkg/objstore"
 )
 
+// UserBucketOverrides defined config overrides supported by UserBucketClient.
+type UserBucketOverrides interface {
```

Comment:
It's implemented by `validation.Overrides`, but we can give it a better name here. (Preferably something without "overrides". What about `TenantKeyProvider`?)

Reply:
I thought to keep it more generic, cause we may need to pass further config overrides in the future (eg. we already need to also support "encryption context" override). I can remove the "overrides" naming an call it `UserBucketConfigProvider` if you prefer.

Reply:
I'm not fan of too generic naming too early, but `UserBucketConfigProvider` is better than `UserBucketOverrides` in my opinion.

Reply:
Discussed offline. We agreed on `TenantConfigProvider`.